packages/firestore/test/integration/api/validation.test.ts

```diff
@@ -1171,6 +1204,36 @@ apiDescribe('Validation:', (persistence: boolean) => {
         "Invalid query. You cannot use 'array-contains' filters with " +
           "'array-contains-any' filters."
       );
+
+      expect(() =>
+        db
+          .collection('test')
+          .where('foo', notInOp, [2, 3])
+          .where('foo', 'array-contains', 1)
+      ).to.throw(
+        "Invalid query. You cannot use 'array-contains' filters with " +
+          "'not-in' filters."
+      );
+    });
+
+    validationIt(persistence, 'with != and NOT_IN filters fail', db => {
+      expect(() =>
+        db
+          .collection('test')
+          .where('foo', notInOp, [2, 3])
+          .where('foo', notEqualOp, 4)
+      ).to.throw(
+        "Invalid query. You cannot use '!=' filters with " + "'not-in' filters."
```

Comment:
What's with the extra plus sign in here? There's a couple in this file. Search for `" + "`.